modules/caddyhttp/matchers.go

```diff
@@ -909,25 +914,51 @@ func (m MatchRemoteIP) getClientIP(r *http.Request) (net.IP, error) {
 	if err != nil {
 		ipStr = remote // OK; probably didn't have a port
 	}
+	// Some IPv6-Adresses can contain zone identifiers at the end,
+	// which are separated with "%"
+	if strings.Contains(ipStr, "%") {
+		zone_id = strings.Split(ipStr, "%")[1]
+		ipStr = strings.Split(ipStr, "%")[0]
```

Comment:
Don't need to run split twice

```suggestion
		split := strings.Split(ipStr, "%")
		ipStr = split[0]
		zoneID = split[1]
```

Reply:
Ahh. I asked myself how it can be done this way. Nice!